modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/util/OrderedMergePublisher.java

```diff
@@ -36,49 +35,52 @@
  *
  * <p>Merges multiple concurrent ordered data streams into one.
  */
-public class SortingCompositePublisher<T> extends CompositePublisher<T> {
+public class OrderedMergePublisher<T> implements Publisher<T> {
     /** Rows comparator. */
-    private final Comparator<T> comp;
+    private final Comparator<? super T> comp;
+
+    /** Array of upstream publishers. */
+    private final Publisher<? extends T>[] sources;
 
     /** Prefetch size. */
     private final int prefetch;
 
     /**
      * Constructor.
      *
-     * @param publishers List of upstream publishers.
      * @param comp Rows comparator.
      * @param prefetch Prefetch size.
+     * @param sources List of upstream publishers.
      */
-    public SortingCompositePublisher(Collection<? extends Publisher<T>> publishers, Comparator<T> comp, int prefetch) {
-        super(publishers);
-
-        this.comp = comp;
+    public OrderedMergePublisher(
+            Comparator<? super T> comp,
+            int prefetch,
+            Publisher<? extends T>... sources) {
+        this.sources = sources;
         this.prefetch = prefetch;
+        this.comp = comp;
     }
 
+    /** {@inheritDoc} */
     @Override
     public void subscribe(Subscriber<? super T> downstream) {
-        subscribe(new OrderedMergeCompositeSubscription<>(downstream, comp, prefetch, publishers.size()), downstream);
+        OrderedMergeSubscription<? super T> subscription = new OrderedMergeSubscription<>(downstream, comp, prefetch, sources.length);
+
+        subscription.subscribe(sources);
+        downstream.onSubscribe(subscription);
+        subscription.drain();
     }
 
     /**
      * Sorting composite subscription.
      *
      * <p>Merges multiple concurrent ordered data streams into one.
      */
-    public static class OrderedMergeCompositeSubscription<T> extends CompositePublisher.CompositeSubscription<T> {
-        /** Marker to indicate completed subscription. */
+    static final class OrderedMergeSubscription<T> implements Subscription {
```

Comment:
```suggestion
    static final class MergedSubscription<T> implements Subscription {
```

Reply:
It seems to me that the current names `OrderedMergePublisher` and `OrderedMergeSubscription` better reflect the purpose of these classes.

Reply:
Concatenated publisher is also ordered, but not sorted, as it iterates over array of upstreams.
Another example, LinkedHashMap is ordered, but not sorted.